pkg/webservice/handlers.go

```diff
@@ -245,7 +245,7 @@ func GetApplicationHistory(w http.ResponseWriter, r *http.Request) {
 		records := imHistory.GetRecords()
 		for _, record := range records {
 			if record == nil {
-				break
+				continue
```

Comment:
The break is correct, it pushes you out of the for loop. Which means you are done collecting records.

The issue is in the return of the function. Leverage the "least indent flow" and early returns to code:
```
	if imHistory == nil {
		http.Error(w, "Internal metrics collection is not enabled.", http.StatusInternalServerError)
		return
	}
	var result []*dao....
	records := ....
	....
}
```

Reply:
No, actually not.

Let's say we have `[<2,3>, <3,4>, nil, nil]` with `pointer=2` in its internal data structure. In the `GetRecords` function first the class copies its array between indexes pointer and the end (`[nil, nil]`), then appends the rest which is the array between 0 to the pointer index (`[<2,3>, <3,4>]`). This results in the following array: `[nil, nil, <2,3>, <3,4>]`. When in the `handler.go#GetApplicationHistory` 's for loop we detect the first nil item in the array (which is the item #0), we break the for loop immediately which is not desired, since we have non-nil entries after that, that we haven't serialized yet.

My original suggestion was to filter the nils in `InternalMetricsCollector` level, because it's more straightforward what to do there. The simplest solution here though to just do this simple fix.